docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+
+## Worker Pools
+
+Now that you configured your Kubernetes cluster control plane in HA, i.e. spread it across multiple zones, you need to do the same for your own workload, but in order to do so, you need to spread your nodes across multiple zones first.
+
+``` yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+spec:
+  provider:
+    workers:
+    - name: ...
+      minimum: 6
+      maximum: 60
+      zones:
+      - ...
+```
+
+Prefer regions with at least 2, better 3+ zones and list the zones in the `zones` section for each of your worker pools. Whether you need 2 or 3 zones at a minimum depends on your fail-over concept:
+
+- Consensus-based software components (like ETCD) depend on maintaining a quorum of `(n/2)+1`, so you need at least 3 zones to tolerate the outage of 1 zone.
+- Primary/Secondary-based software components need just 2 zones to tolerate the outage of 1 zone.
+- Then there are software components that can scale out horizontally. They are probably fine with 2 zones, but you also need to think about the load-shift and that the remaining zone must then pick up the work of the unhealthy zone. With 2 zones, the remaining zone must cope with an increase of 100% load. With 3 zones, the remaining zones must only cope with an increase of 50% load (per zone).
+
+In general, the question is also whether you have the fail-over capacity already up and running or not. If not, i.e. you depend on re-scheduling on a healthy zone or auto-scaling, be aware that during a zone outage, you will see a resource crunch in the healthy zones. If you have no automation, i.e. only human operators (a.k.a. "red button approach"), you probably will not get the machines you need and even with automation, it may be tricky. But holding the capacity available at all times is costly. In the end, that's a decision only you can make. If you made that decision, please adapt the `minimum` and `maximum` settings for your worker pools accordingly.
+
+Also, consider fall-back worker pools (with different/alternative machine types) and cluster autoscaler expanders, too. Read more about those concepts [here](https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/FAQ.md#what-are-expanders) and [here](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/expander/priority/readme.md).
+
+Gardener-managed clusters deploy the [cluster autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) or CA for short and you can tweak the general CA knobs for Gardener-managed clusters like so:
+
+``` yaml
```

Comment:
Actually, I did that on purpose in many places, not as in-place documentation (not at all), but so that people see it/get the idea what can be done. People will not always click the links and may then not see the (useful) options offered there. So, it places the information more directly in front of them. I at least like that when I read such papers, i.e. I like it if the author lets me glimpse into the space of options and seeds ideas what can be done rather than only passing me on to the docs.